src/core/solvers/ipopt/ipopt-iface.cpp

```diff
@@ -0,0 +1,665 @@
+  std::size_t ixu = 0;
+  for (std::size_t t = 0; t < T_; t++) {
+    const std::size_t ndxi = models[t]->get_state()->get_ndx();
+    const std::size_t nui = models[t]->get_nu();
+
+    for (std::size_t j = 0; j < ndxi; j++) {
+      x[ixu + j] = 0;
+    }
+
+    for (std::size_t j = 0; j < nui; j++) {
+      x[ixu + ndxi + j] = us_[t](j);
+    }
+
+    ixu += ndxi + nui;
+  }
+
+  // Terminal state
+  {
+    const boost::shared_ptr<ActionModelAbstract> &model = problem_->get_terminalModel();
+    const std::size_t ndxi = model->get_state()->get_ndx();
+    for (std::size_t j = 0; j < ndxi; j++) {
+      x[ixu + j] = 0;
+    }
+  }
+
+  return true;
+}
+
+bool IpoptInterface::eval_f(Ipopt::Index n, const Ipopt::Number *x, bool new_x, Ipopt::Number &obj_value) {
+  assert_pretty(n == nvar_, "Inconsistent number of decision variables");
+
+  // Running costs
+  std::size_t ixu = 0;
+  for (std::size_t t = 0; t < T_; t++) {
+    const boost::shared_ptr<ActionModelAbstract> &model = problem_->get_runningModels()[t];
+    const boost::shared_ptr<ActionDataAbstract> &data = problem_->get_runningDatas()[t];
+    const std::size_t ndxi = model->get_state()->get_ndx();
+    const std::size_t nui = model->get_nu();
+
+    datas_[t]->dx = Eigen::VectorXd::Map(x + ixu, ndxi);
+    datas_[t]->u = Eigen::VectorXd::Map(x + ixu + ndxi, nui);
+
+    model->get_state()->integrate(xs_[t], datas_[t]->dx, datas_[t]->x);
+
+    model->calc(data, datas_[t]->x, datas_[t]->u);
+
+    obj_value += data->cost;
+
+    ixu += ndxi + nui;
+  }
+
+  {
+    // Terminal costs
+    const boost::shared_ptr<ActionModelAbstract> &model = problem_->get_terminalModel();
+    const boost::shared_ptr<ActionDataAbstract> &data = problem_->get_terminalData();
+    const std::size_t ndxi = model->get_state()->get_ndx();
+
+    datas_[T_]->dx = Eigen::VectorXd::Map(x + ixu, ndxi);
+    model->get_state()->integrate(xs_[T_], datas_[T_]->dx, datas_[T_]->x);
+
+    model->calc(data, datas_[T_]->x);
+    obj_value += data->cost;
+  }
+
+  return true;
+}
+
+bool IpoptInterface::eval_grad_f(Ipopt::Index n, const Ipopt::Number *x, bool new_x, Ipopt::Number *grad_f) {
+  assert_pretty(n == nvar_, "Inconsistent number of decision variables");
+
+  std::size_t ixu = 0;
+
+  for (std::size_t t = 0; t < T_; t++) {
+    const boost::shared_ptr<ActionModelAbstract> &model = problem_->get_runningModels()[t];
+    const boost::shared_ptr<ActionDataAbstract> &data = problem_->get_runningDatas()[t];
+    const std::size_t ndxi = model->get_state()->get_ndx();
+    const std::size_t nui = model->get_nu();
+
+    datas_[t]->dx = Eigen::VectorXd::Map(x + ixu, ndxi);
+    datas_[t]->u = Eigen::VectorXd::Map(x + ixu + ndxi, nui);
+
+    model->get_state()->integrate(xs_[t], datas_[t]->dx, datas_[t]->x);
+    model->get_state()->Jintegrate(xs_[t], datas_[t]->dx, datas_[t]->Jsum_dx, datas_[t]->Jsum_dx, second, setto);
+
+    model->calc(data, datas_[t]->x, datas_[t]->u);
```

Comment:
Are you sure we need to run `calc`? it seems improbable to me that Ipopt needs this.

Reply:
It does need this call. We need this calc here since this function is the first to be called to compute the search direction.

Removing this call the solver exits without running a single iteration.